crates/xmtpv3/src/session.rs

```diff
@@ -0,0 +1,34 @@
+use anyhow::Result;
+use vodozemac::olm::{OlmMessage, Session};
+
+// Wrapper struct for Olm sessions with additional metadata
+pub struct VoodooSession {
+    pub session: Session,
+    // NOTE: this does not save the first plaintext encoded in the
+    // outbound session
+    // TODO: instead of storing pure plaintext, we should store higher level structs
+    pub my_messages: Vec<String>,
+    pub their_messages: Vec<String>,
+}
+
+impl VoodooSession {
+    pub fn new(olm_session: Session) -> Self {
+        Self {
+            session: olm_session,
+            my_messages: Vec::new(),
+            their_messages: Vec::new(),
```

Comment:
How might this look like in the final version? Would there still be a need to store plaintexts on the Session or is this to facilitate the prototype only?

Reply:
Great question: I'm open to however we want to structure this. We're going to be exporting both session state and message history from Rust land at some point right? (or not - asking to gut-check my assumptions)